From e2b56e7d8786f7000c3159346b6cf2360ad3c8c3 Mon Sep 17 00:00:00 2001
Message-Id: <e2b56e7d8786f7000c3159346b6cf2360ad3c8c3.1641573384.git.stefan@agner.ch>
In-Reply-To: <9d68dedb813f942286856c9d8c2eed01b55bb473.1641573384.git.stefan@agner.ch>
References: <9d68dedb813f942286856c9d8c2eed01b55bb473.1641573384.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 4 Mar 2021 14:33:09 +0100
Subject: [PATCH 2/9] ARM: dts: bcm2711: yellow: Mux UART4 for SiLabs radio
 module

Enable UART4 by default and mux pins including hardware flow control.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 arch/arm/boot/dts/bcm2711-rpi-cm4-ha-yellow.dts | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/bcm2711-rpi-cm4-ha-yellow.dts b/arch/arm/boot/dts/bcm2711-rpi-cm4-ha-yellow.dts
index 5d4e058cac43..6b702d66ac55 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-cm4-ha-yellow.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-cm4-ha-yellow.dts
@@ -21,6 +21,7 @@ memory@0 {
 	};
 
 	aliases {
+		serial4 = &uart4;
 		emmc2bus = &emmc2bus;
 		ethernet0 = &genet;
 		pcie0 = &pcie0;
@@ -321,6 +322,14 @@ &uart1 {
 	status = "okay";
 };
 
+/* uart4 for Zigbee */
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart4_pins>;
+	uart-has-rtscts;
+	status = "okay";
+};
+
 &vchiq {
 	interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
 };
@@ -549,9 +558,9 @@ uart3_pins: uart3_pins {
 	};
 
 	uart4_pins: uart4_pins {
-		brcm,pins = <8 9>;
+		brcm,pins = <8 9 10 11>;
 		brcm,function = <BCM2835_FSEL_ALT4>;
-		brcm,pull = <0 2>;
+		brcm,pull = <0 2 2 0>;
 	};
 
 	uart5_pins: uart5_pins {
-- 
2.34.1

